docs/LlpcOverview.md

```diff
@@ -0,0 +1,71 @@
+# LLPC overview
+
+This document outlines the state of LLPC as it will be at the end of the
+"middle-end interface clean-up" series of patches.
+
+## Compilation process
+
+The LLPC compilation process consists of three phases:
+* front-end;
+* middle-end (LGC);
+* back-end (LLVM AMDGPU back-end).
+
+### Front-end
+
+* Translate the input SPIR-V to LLVM IR suitable for LGC.
+* Run some "lowering" passes to help with that transformation.
+* This phase calls the LGC (middle-end) interface to help with:
+  - building LLVM IR;
+  - setting state;
+  - linking the individual shader IR modules into a single pipeline IR module.
+* All the SPIR-V and Vulkan specific handling is within the front-end.
+
+After this, when using LGC's Builder in its BuilderRecorder mode, as LLPC does, the
+pipeline IR module contains calls to `llpc.call.*` functions that record the Builder
+methods used by the front-end. The IR module also contains metadata representing the
+pipeline state.
+
+### Middle-end (LGC)
+
+See
+[LGC (LLPC middle-end) overview](LgcOverview.md)
+for more details.
+
+* In BuilderRecorder mode, run the BuilderReplayer pass to lower the `llpc.call.*` calls
+  into actual IR, and read the pipeline state metadata.
+* Perform further lowering on the LLVM IR to obtain LLVM IR with `llvm.amdgcn.*` intrinsics
+  understood by the AMDGPU back-end, and the PAL metadata needed by the back-end to
+  add to the ELF.
+* Run middle-end optimizations, most of which are standard LLVM ones.
```

Comment:
Could you high-light some special works in LGC, include NGG, merge shader, assign shader register value and etc in some place, either in this doc or in LgcOverview.md. thanks!

Reply:
Good idea, I will do.